pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/store/TableViewLoadDataStoreImpl.java

```diff
@@ -150,9 +156,34 @@ public synchronized void init() throws IOException {
         start();
     }
 
-    private synchronized void validateTableViewStart() {
-        if (tableView == null) {
-            throw new IllegalStateException("table view has not been started");
+    private void validateProducer() {
+        if (producer == null || !producer.isConnected()) {
+            try {
+                if (producer != null) {
+                    producer.close();
+                }
+                producer = null;
+                startProducer();
+                log.info("Restarted producer on {}", topic);
+            } catch (Exception e) {
+                log.error("Failed to restart producer on {}", topic, e);
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private void validateTableView() {
+        if (tableView == null || System.currentTimeMillis() - tableViewLastUpdateTimestamp
+                > ((long) conf.getLoadBalancerReportUpdateMaxIntervalMinutes()) * 60 * 1000 * 2) {
+            tableViewLastUpdateTimestamp = 0;
+            try {
+                closeTableView();
+                startTableView();
+                log.info("Restarted tableview on {}", topic);
```

Comment:
Should we add more info about why we restart?

Reply:
updated.